robocop/checkers/misc.py

```diff
@@ -601,6 +607,78 @@
 
         """,
     ),
+    "0927": Rule(
+        RuleParam(
+            name="sections_order",
+            default="documentation,tags,timeout,setup,template,keyword,teardown",
+            converter=parse_test_case_order_param,
+            show_type="str",
+            desc="order of sections in comma-separated list",
+        ),
+        rule_id="0927",
+        name="test-case-section-out-of-order",
+        msg="'{{ section_name }}' is in wrong place of Test Case. "
+        "Recommended order of elements in Test Cases: {{ recommended_order }}",
+        severity=RuleSeverity.WARNING,
+        added_in_version="5.3.0",
+        docs="""
+        Sections should be defined in order set by ``sections_order``
+        parameter (default: ``documentation,tags,timeout,setup,template,keyword,teardown``).
+
+        To change the default order use following option::
+
+            robocop --configure test-case-section-out-of-order:sections_order:comma,separated,list,of,sections
+
+        where section should be case-insensitive name from the list: 
+        documentation, tags, timeout, setup, template, keywords, teardown. 
+        Order of not configured sections is ignored.
+    
+        Example of rule violation::
+
+            *** Test Cases ***
+            Keyword After Teardown
+                [Documentation]    This is test Documentation
+                [Tags]    tag1    tag2
+                [Teardown]    Log    abc
+                Keyword1
+        """,
+    ),
+    "0928": Rule(
+        RuleParam(
+            name="sections_order",
+            default="documentation,tags,arguments,timeout,setup,keyword,teardown",
```

Comment:
It may be now obvious when you did it, but I like that you're using single order for before / body / after. In Robotidy for corresponding transformer that does order the settings I have before / after parameters. It would be better to just use single parameter like you did - I may do it in the future